accepted/0000-hub-online-synchronization.md

```diff
@@ -0,0 +1,267 @@
+- Cross organization synchronization
+
+### ISSv2
+
+Cons:
+- Slow to export and import
+   - Data transfer is not optimized: uses sql statements with a single transaction. The bigger the channel or more channels in one single export, the bigger the problem.
+- If database schema doesn't follow the conventions, we need manual corrections
+- A lot of manual steps to run and make it more efficient
+- No UI support
+- Hard to debug
+- No good feedback during import
+- No cross organization import (organizations needs to have the same name in source and target)
+- Need to have all the cloned channels hierarchy to be able to use SP migration
+
+Pros:
+- Can transfer more than just software channels (images and configuration channels)
+- Sync process is started from the server side
+- Export once, import in parallel in all peripherals
+- Support for fully disconnected environment
+
+
+## Proposed solution
+
+One of SUSE Manager focus areas is synchronizing and managing content, especially RPM's. To do so, it synchronizes remote channels, apply filters to produce new channels (CLM) and makes these channels available to be consumed.
+
+The overall idea of the Hub Online synchronization is to re-use the existing repo-sync mechanism and synchronize channels in the peripheral from the HUB published repositories (online connection between hub and peripherals servers).
+
+In the first stage the focus will be synchronizing software channels, since it's the most problematic one in terms of performance and automation. However, we should be able to re-use the integration mechanism, since everything will be implemented through API.
+
+We will need a mechanism to create the channels in the peripheral servers (vendor and customs ones) in the desired organization. Those channels need to have as content source the corresponding HUB server repository URL, with the correct authentication token. With this, repo-sync will take care of channel synchronization (proof of concept already tested).
+
+Pros:
+- Automatically update channel during the night with repo-sync/mgr-sync
+- No need to have specific code adaptations in case of schema changes. It will be handled any way in repo-sync changes
+- Rely on a stable specification: repositories metadata
+- After initial configuration, all synchronization between hub and peripheral should be automatic and transparent
+  - Only differences would be transferred
+- Promote environments in CLM will automatically propagate changes to all connected peripherals
+- Automatic creating of bootstrap repository done by repo-sync tool
+
+Cons:
+- All peripheral can start synchronizing at same time
+  - Can be problematic if we have several peripherals performing full synchronization at same time. However we can configure the peripherals to run repo-sync in different hours and spread the load
+  - This is only a problem in first sync, since subsequent sync only transfer differences
+
```

Comment:
When synchronizing from SCC service, in theory, we rely on a service with HA.
In that proposal, the peripherals will rely in a unique Hub instance, what happen if that goes down or.. the Hub disk burns? I would also consider how we can recover under these cases.

Reply:
@srbarrios is this stilla question for you?